app/service/login_handlers/default.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import ldap3
+
+from aiohttp import web
+from aiohttp_jinja2 import render_template
+from ldap3.core.exceptions import LDAPAttributeError, LDAPException
+
+from app.service.interfaces.i_login_handler import LoginHandlerInterface
+
+HANDLER_NAME = 'Default Login Handler'
+
+
+class DefaultLoginHandler(LoginHandlerInterface):
+    def __init__(self, services):
+        super().__init__(HANDLER_NAME)
+        self.log = logging.getLogger('default_login_handler')
+        self.auth_svc = services.get('auth_svc')
```

Comment:
It calls the auth svc's `provide_verified_login_response` upon successful login. But I can have it access the auth_svc during that time and not during `__init__`

Reply:
Deferring the auth service lookup  until its needed is something to consider as a means to avoid problems with initialization order (e.g., this is initialized before the auth svc instance has been added to the service registry) but has the drawback of hiding dependencies.

I was getting at the fact that `.get()` should not be used if it's not an optional dependency. Using `.get()` will set it to `None`, which means it'll just blow up in a strange way later during an authentication flow. If it's not truly optional , don't use `.get()` (or use `.get()` with an appropriate fallback that works).

Reply:
Gotcha, I have it throw an exception now if the auth service isn't found